csrc/device_lower/pass/insert_syncs.cpp

```diff
@@ -39,7 +39,13 @@ class SmemAllocMap {
   //! Insert a new node if it's a SMEM allocation
   void insert(kir::Allocate* alloc) {
     if (auto tv = dynamic_cast<TensorView*>(alloc->buffer())) {
-      if (tv->getMemoryType() == MemoryType::Shared) {
+      // Tensor memory is similar to shared memory because they are both
+      // shared between threads in a block. In that sense, we can consider
+      // tensor memory as special type of shared memory. In this file, we use
+      // the term "shared memory", "smem" to refer to both shared and tensor
+      // memories.
+      if (tv->getMemoryType() == MemoryType::Shared ||
+          tv->getMemoryType() == MemoryType::Tensor) {
```

Comment:
nit: why not adding something like:

```
bool isSharedMemory(TensorView* tv ){
  return tv->getMemoryType() == MemoryType::Shared ||
          tv->getMemoryType() == MemoryType::Tensor;
}
```

Then, the changes of this PR would be just calling this function.

Reply:
changed